crates/db_views/src/post_view.rs

```diff
@@ -352,40 +352,44 @@ fn queries<'a>() -> Queries<
       query = query.filter(post_aggregates::creator_id.eq(creator_id));
     }
 
-    if let Some(person_id) = person_id {
-      let is_subscribed = exists(
-        community_follower::table.filter(
-          post_aggregates::community_id
-            .eq(community_follower::community_id)
-            .and(community_follower::person_id.eq(person_id)),
-        ),
-      );
-      match options.listing_type.unwrap_or_default() {
-        ListingType::Subscribed => query = query.filter(is_subscribed),
-        ListingType::Local => {
-          query = query
-            .filter(community::local.eq(true))
-            .filter(community::hidden.eq(false).or(is_subscribed));
-        }
-        ListingType::All => query = query.filter(community::hidden.eq(false).or(is_subscribed)),
-        ListingType::ModeratorView => {
-          query = query.filter(exists(
-            community_moderator::table.filter(
-              post::community_id
-                .eq(community_moderator::community_id)
-                .and(community_moderator::person_id.eq(person_id)),
-            ),
-          ));
+    if let Some(listing_type) = options.listing_type {
+      if let Some(person_id) = my_person_id {
+        let is_subscribed = exists(
+          community_follower::table.filter(
+            post_aggregates::community_id
+              .eq(community_follower::community_id)
+              .and(community_follower::person_id.eq(person_id)),
+          ),
+        );
+        match listing_type {
+          ListingType::Subscribed => query = query.filter(is_subscribed),
+          ListingType::Local => {
+            query = query
+              .filter(community::local.eq(true))
+              .filter(community::hidden.eq(false).or(is_subscribed));
+          }
+          ListingType::All => query = query.filter(community::hidden.eq(false).or(is_subscribed)),
+          ListingType::ModeratorView => {
+            query = query.filter(exists(
+              community_moderator::table.filter(
+                post::community_id
+                  .eq(community_moderator::community_id)
+                  .and(community_moderator::person_id.eq(person_id)),
+              ),
+            ));
+          }
         }
       }
-    } else {
-      match options.listing_type.unwrap_or_default() {
-        ListingType::Local => {
-          query = query
-            .filter(community::local.eq(true))
-            .filter(community::hidden.eq(false));
+      // If your person_id is missing, only show local
+      else {
+        match listing_type {
+          ListingType::Local => {
+            query = query
+              .filter(community::local.eq(true))
+              .filter(community::hidden.eq(false));
+          }
+          _ => query = query.filter(community::hidden.eq(false)),
         }
```

Comment:
For the `list_posts` endpoint, the listing_type will always be defined, and it takes into account the `default_listing_type` : 

https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/api/list_posts.rs#L48

I don't know that its necessary to add a unit test for that one, because the default_listing_type is at the API level anyway.

Reply:
Then maybe we can make the listing type param mandatory here instead of an option?

Reply:
I could go either way on that, but I think its nice to have it be optional. I'll address the hidden thing in a sec.